backend/src/AppServer.hs

```diff
@@ -174,9 +177,33 @@ getLeaderboardHandler year =
 adminRenamePlayerHandler :: RenamePlayerRequest -> AppM NoContent
 adminRenamePlayerHandler RenamePlayerRequest {pid, newName} = runDb $ updatePlayerName pid newName >> pure NoContent
 
+adminModifyReportHandler :: ModifyReportRequest -> AppM NoContent
+adminModifyReportHandler ModifyReportRequest {rid, report} = case validateReport report of
+  Failure errors -> throwError $ err422 {errBody = show errors}
+  Success _ -> runDb $ do
+    oldReport <- readOrError ("Cannot find report " <>: rid) $ lift . get $ rid
+    Entity newWinnerId _ <- readOrError ("Cannot find player " <>: report.winner) $ getPlayerByName report.winner
+    Entity newLoserId _ <- readOrError ("Cannot find player " <>: report.loser) $ getPlayerByName report.loser
+
+    let newReport = toGameReport oldReport.gameReportTimestamp newWinnerId newLoserId report
```

Comment:
Note: the timestamp is unmodifiable, currently. We can change that in the future, if we want.

Reply:
🕥